clients/drcachesim/tools/view.cpp

```diff
@@ -109,6 +109,19 @@ view_t::process_memref(const memref_t &memref)
 
     if (memref.marker.type == TRACE_TYPE_MARKER) {
         switch (memref.marker.marker_type) {
+        case TRACE_MARKER_TYPE_FILETYPE:
+            if (TESTANY(OFFLINE_FILE_TYPE_ARCH_ALL, memref.marker.marker_value) &&
+                !TESTANY(IF_X86_ELSE(IF_X64_ELSE(OFFLINE_FILE_TYPE_ARCH_X86_64,
+                                                 OFFLINE_FILE_TYPE_ARCH_X86_32),
+                                     IF_X64_ELSE(OFFLINE_FILE_TYPE_ARCH_AARCH64,
+                                                 OFFLINE_FILE_TYPE_ARCH_ARM32)),
+                         memref.marker.marker_value)) {
+                error_string_ = std::string("Architecture mismatch: trace recorded on ") +
```

Comment:
It's worth clarifying that this error string is about bitwidth mismatch by adding what the expected architecture/bitwidth should be. It took me awhile to parse those macros.

Reply:
OK I added the architecture the tool was compiled for to the output.